### PR TITLE
PD: Improve InvoluteGear's fillets

### PR DESCRIPTION
**Pull request by jbaehr**
Previously, the fillet started at a fixed value, tailored at 20° pressure angle, a fillet radius of 0.375 and a dedendum of 1.25. With PR #8184 those values became user-adjustable and thus the assumptions of the original code are not fullfilled any more. This resulted in significant artefacts, especially for higher pressure angles or smaller dedendums as commonly found in splines.

This commit actually calculates the junction between fillet and involute so that the tangents of the both curves match. The mathematical approach is described in source code comments.

The change in the fillet calculation also affects compatibility with files generated by earlier versions of FreeCAD. Those changes are way below 0.1% per tooth, however the earlier test required absolute equality down to the micron. This was relaxed and also changed to a relative, per tooth, tolerance.

There is one particular case where the new algorithm performs slightly worse, though. That is when the fillet radius is too large to the dedendum, i.e. resulting in a single arc instead of two fillets, and effectively cannibalizes some of the clearance. This happens with internal gears having less than 25 teeth. At about 15 teeth it becomes visible that the fillet is not 100% tanget any more. However, as such a low number of teeth is highly unusual for internal gears and the effect,
although noticeable, is minor, the overall benefits of the new algorithm outweighs the drawbacks. And now with the fillet radius being adjustable it is easy to fix, too.  
The technical reason is that the tangency is calculated correctly, but the fillet circle is displaced afterwards to avoid an overlap of the two fillets. For the new fillet position, the tangets do not align any more.

image
image

----

Strictly speaking the changes to the standalone svg exporter are independent of the fillet generation. I just made them during my work on the fillets for allow for faster visual inspection. If you want, I can split this off into a dedicated PR.

----

- [x]  Your Pull Request meets the requirements outlined in section 5 of [CONTRIBUTING.md](https://github.com/FreeCAD/FreeCAD/blob/master/CONTRIBUTING.md) for a Valid PR

Please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [1.0 Changelog Forum Thread](https://forum.freecad.org/viewtopic.php?f=10&t=69438).

---

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8424">pipeline status</a> for feature branch [PR_8424](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8424). Pipeline [774463569](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/774463569) was triggered at [1d49549](https://github.com/FreeCAD/FreeCAD/pull/8424/commits/1d495497bf9ac49e084bbf5da197b68b7534b3e7). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8424">pipeline status</a> for feature branch [PR_8424](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8424). Pipeline [774477012](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/774477012) was triggered at [23cc65d](https://github.com/FreeCAD/FreeCAD/pull/8424/commits/23cc65da793448d62ef49ed14b56d0978e5ff8f2). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/PartDesign/fcgear/svggear.py [hunk 1]
```diff
@@ -57,14 +52,64 @@ def curve(self, *points):
...
 
+
 if __name__ == '__main__':
     from optparse import OptionParser
-    p = OptionParser()
-    p.add_option('-a', '--angle', help='pressure angle',
-                 dest='angle', default=20)
+    p = OptionParser(
+        usage="usage: %prog [options] [[MODULE] NUMER_OF_TEETH]",
+        description=("Generates the outline of a metric, involtue gear. "
```
Comment (luzpaz) [inline src/Mod/PartDesign/fcgear/svggear.py:60]: involtue->involute

Comment (jbaehr) [reply src/Mod/PartDesign/fcgear/svggear.py]: thx, fixed. This has even slippet through our CI's linter.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8424">pipeline status</a> for feature branch [PR_8424](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8424). Pipeline [774733327](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/774733327) was triggered at [b4aaebd](https://github.com/FreeCAD/FreeCAD/pull/8424/commits/b4aaebd451f021ddb1a0273df391761f77a9eace). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/PartDesign/fcgear/involute.py [hunk 2]
```diff
@@ -58,23 +59,64 @@ def CreateExternalGear(w, m, Z, phi,
...
+        # For a circle in parameter form, the tangent angle is pi/2 - t (for the first quadrant,
+        # i.e. 0 <= t <= pi/2, and more than one quadrant is not of interest for us). Unfortunately,
+        # the fillet circle is offset from the origin by Rc/phi, so things becomes more messy:
+        # rc(t) = sqrt((Rc*cos(phi) + fRad*cos(t))**2 + (Rc*sin(phi) + fRad*sin(t))**2)
+        # To get rid or any sin^2 and cos^2 we assume phi "very small", i.e. sin(phi) becomes 0.
+        # This is justyfied because Rc is much larger than fRad and the parallax error is
+        # neglectable. Next we substitute the parameter t by pi/2-q-pi. For one to account for the
+        # tangent angle definitions (see above), and then to turn our cicle by 180° as for the
+        # root fillet we need the third quadrant of the circle (NB: we are looking at the upper
+        # half of the right most tooth, i.e. the involute gorws downwards!). This results in
```
Comment (luzpaz) [inline src/Mod/PartDesign/fcgear/involute.py:84]: gorws->grows

Comment (jbaehr) [reply src/Mod/PartDesign/fcgear/involute.py]: Thanks again! I read over this so many times that by now I got completely blind ;-)

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8424">pipeline status</a> for feature branch [PR_8424](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8424). Pipeline [774859809](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/774859809) was triggered at [4d14c41](https://github.com/FreeCAD/FreeCAD/pull/8424/commits/4d14c4119655ecb643acb9dcf12f47d663fd6a03). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Pull request merged**